src/lepton/jpgcoder.cc

```diff
@@ -2004,9 +2004,17 @@ void process_file(IOUtil::FileReader* reader,
     }
     if (!fast_exit) {
         // close iostreams
-        if ( str_in  != NULL ) delete( str_in  ); str_in  = NULL;
-        if ( str_out != NULL ) delete( str_out ); str_out = NULL;
-        if ( ujg_out != NULL ) delete( ujg_out ); ujg_out = NULL;
+        if (ujg_base_in != str_in) {
```

Comment:
```cpp
// str_in may be the same as ujg_base_in or Sirikata::BufferedReader(ujg_base_in)
// In the first case we should reset ujg_base_in to avoid double delete of the same memory.
if (ujg_base_in == str_in) {
    ujg_base_in = NULL;
}
delete( str_in  ); str_in = NULL;
delete( ujg_base_in );  ujg_base_in = NULL;
```
Note that I prefer to delete str_in first since it may (theoretically) perform operations on ujg_base_in.

Reply:
That's okay too